velox/functions/prestosql/Split.cpp

```diff
@@ -0,0 +1,263 @@
+      }
+
+      // Ensure that our result elements vector uses the same string buffer as
+      // the input vector of strings.
+      builder.setStringBuffers({strings->base()
+                                    ->as<ConstantVector<StringView>>()
+                                    ->getStringBuffer()});
+    } else {
+      // The rest of the cases are handled through this general path and no
+      // direct access.
+      applyDecoded<I>(rows, strings, delims, limits, builder);
+
+      // Ensure that our result elements vector uses the same string buffer as
+      // the input vector of strings.
+      builder.setStringBuffers(strings->base());
+    }
+
+    std::shared_ptr<ArrayVector> arrayVector =
+        std::move(builder).finish(context->pool());
+    context->moveOrCopyResult(arrayVector, rows, result);
+  }
+
+  template <typename I>
+  void applyDecoded(
+      const SelectivityVector& rows,
+      DecodedVector* strings,
+      DecodedVector* delims,
+      DecodedVector* limits,
+      ArrayBuilder<Varchar>& builder) const {
+    if (limits == nullptr) {
+      auto limit = std::numeric_limits<I>::max();
+      rows.applyToSelected([&](vector_size_t row) {
+        applyInner<false, I>(
+            strings->valueAt<StringView>(row),
+            delims->valueAt<StringView>(row),
+            limit,
+            row,
+            builder);
+      });
+    } else {
+      rows.applyToSelected([&](vector_size_t row) {
+        applyInner<true, I>(
+            strings->valueAt<StringView>(row),
+            delims->valueAt<StringView>(row),
+            limits->valueAt<I>(row),
+            row,
+            builder);
+      });
+    }
+  }
+
+  /**
+   * The inner most kernel of the vector operations for 'split'.
+   */
+  template <bool hasLimit = false, typename I>
+  inline void applyInner(
+      StringView input,
+      const StringView delim,
+      I limit,
+      vector_size_t row,
+      ArrayBuilder<Varchar>& builder) const {
+    // Add new array (for the new row) to our array vector.
+    auto arrayRef = builder.startArray(row);
+
+    // Trivial case of converting string to array with 1 element.
+    if (hasLimit and limit == 1) {
+      arrayRef.emplace_back(input);
+      return;
+    }
+
+    // We walk through our input cutting off the pieces using the delimiter and
+    // adding them to the elements vector, until we reached the end of the
+    // string or the limit.
+    int32_t addedElements{0};
+    std::string_view sinput(input.data(), input.size());
+    const std::string_view sdelim(delim.data(), delim.size());
+    while (true) {
+      // Find the byte of the 1st delimiter.
+      auto byteIndex = sinput.find(sdelim, 0);
+
+      // Delimiter is not found, leave the loop.
+      if (byteIndex == std::string_view::npos) {
+        break;
+      }
+
+      // Add the new element, we've split
+      arrayRef.emplace_back(StringView(sinput.data(), byteIndex));
```

Comment:
You don't need to call the `StringView` constructor when using `emplace`.
```suggestion
      arrayRef.emplace_back(sinput.data(), byteIndex);
```